packages/theme/src/typography.ts

```diff
@@ -1,26 +1,100 @@
-        fontFamily: fontFamilies.base,
-    },
-    title2: {
-        fontSize: fontSizes.title2,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
-    },
-    title3: {
-        fontSize: fontSizes.title3,
-        lineHeight: lineHeights.small,
-        fontWeight: fontWeights.regular,
-        fontFamily: fontFamilies.base,
+export const fontWeights = {
+    medium: 400,
+    demi: 600,
+} as const;
+
+export type FontWeight = keyof typeof fontWeights;
+export type FontWeightValue = typeof fontWeights[FontWeight];
+
+// we need unit-less typography base because RN is unit-less, we can easily add units later
+// for web we need string instead of object because styled-components syntax
+export const typographyStylesBase = {
+    largeTitle: {
+        fontSize: 48,
+        lineHeight: 53,
+        fontWeight: fontWeights.medium,
+        letterSpacing: 0.4,
+    },
+    mediumTitle: {
+        fontSize: 34,
+        lineHeight: 37,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -1.4,
+    },
+    smallTitle: {
+        fontSize: 22,
+        lineHeight: 26,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    highlight: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.4,
+    },
+    body: {
+        fontSize: 16,
+        lineHeight: 24,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.4,
+    },
+    callout: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.demi,
+        letterSpacing: -0.3,
+    },
+    hint: {
+        fontSize: 14,
+        lineHeight: 21,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.3,
+    },
+    label: {
+        fontSize: 12,
+        lineHeight: 18,
+        fontWeight: fontWeights.medium,
+        letterSpacing: -0.1,
     },
 } as const;
+
+export type TypographyStyle = keyof typeof typographyStylesBase;
+export type TypographyStyles = Record<TypographyStyle, string>;
+export type NativeTypographyStyleDefinition = {
+    fontSize: number;
+    lineHeight: number;
+    fontWeight: FontWeightValue;
+    letterSpacing: number;
+    fontFamily?: string;
+};
+export type NativeTypographyStyles = Record<TypographyStyle, NativeTypographyStyleDefinition>;
+
+export const prepareTypography = (): TypographyStyles => {
+    const typography: any = {};
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [styleName, value] of Object.entries(typographyStylesBase)) {
+        // for tagged template styles on web we need just string snippet
+        typography[styleName] = `
+            font-size: ${value.fontSize}px;
+            line-height: ${value.lineHeight}px;
+            font-weight: ${value.fontWeight};
+            letter-spacing: ${value.letterSpacing}px;
+            font-family: ${fontFamilies.base}
```

Comment:
kinda redundant having the family in all of them, no? Also so far we have only 1 Font, so I imagine setting a global font for a theme and also export all other fonts when they are added to use them in specific places.

Reply:
I expect these part of code to be tuned up when someone will start using it. 

I not sure what will be plan for migration to new theme in web env, because if we use styled-components, it will be paint in the ass because you will have to support like 2 themes in styled-components and you probably can't migrate everything at one time, but you should that do gradually, so you will maybe start using some components one by one from new theme and then this could be useful because of scope. Also could be useful in storybook I think.

I think it won't cause any harm if it will be there and when someone will use it in web and will think it's not necessary, it will be pretty easy to remove.

Reply:
ok, I am deleting that :D

Reply:
done